oneanddone/tasks/static/js/tasklist.js

```diff
@@ -0,0 +1,415 @@
+        <TaskList data={this.state.data} onLoadMore={this.loadMoreTasks} moreTasks={this.state.moreTasks} />
+      </div>
+    );
+  }
+});
+
+let TaskFilters = React.createClass({
+  handleSearchChange: function(e) {
+    this.props.filters.search = e.target.value;
+    this.props.onFilterChange();
+  },
+  render: function() {
+    return (
+      <div className="task-filters">
+        <form method="get" className="filters">
+          <fieldset>
+            <SearchFilter {...this.props} />
+            <ETFilter {...this.props} />
+            <TeamFilter {...this.props} />
+            <ProjectFilter {...this.props} />
+            <TypeFilter {...this.props} />
+          </fieldset>
+        </form>
+      </div>
+    );
+  }
+});
+
+let SearchFilter = React.createClass({
+  handleChange: function(e) {
+    this.props.filters.search = e.target.value;
+    this.props.onFilterChange();
+  },
+  render: function() {
+    return (
+      <p>
+        <input type="text" id="search" placeholder="Search for tasks" onChange={this.handleChange} />
+      </p>
+    );
+  }
+});
+
+let ETFilter = React.createClass({
+  render: function() {
+    let etNodes = [15, 30, 45, 60].map(function(executionTime) {
+      return (
+        <ET executionTime={executionTime}
+            key={executionTime}
+            {...this.props} />
+      );
+    }, this);
+    return (
+      <div className="et-filter">
+        <p>Estimated minutes to complete:</p>
+        <p>
+          {etNodes}
+        </p>
+      </div>
+    );
+  }
+});
+
+let ET = React.createClass({
+  handleChange: function(e) {
+    let executionTime = this.props.filters.execution_time;
+    if (e.target.checked) {
+      executionTime.push(e.target.value);
+    } else {
+      executionTime.splice($.inArray(e.target.value, executionTime), 1);
+    }
+    this.props.onFilterChange();
+  },
+  render: function() {
+    let etId = 'et' + this.props.executionTime;
+    return (
+      <label htmlFor={etId}>
+        <input id={etId} value={this.props.executionTime} type="checkbox" onClick={this.handleChange} />
+        {this.props.executionTime}
+      </label>
+    );
+  }
+});
+
+let TeamFilter = React.createClass({
+  getInitialState: function() {
+    return {
+      expanded: true
+    }
+  },
+  handleToggle: function(e) {
+    this.setState({expanded: !this.state.expanded});
+  },
+  render: function () {
+    let teamNodes = this.props.teams.map(function(team) {
+      if (this.state.expanded || $.inArray(team.id.toString(), this.props.filters.team) > -1) {
+        return (
+          <Team id={team.id}
+                name={team.name}
+                url={team.url}
+                key={team.id}
+                {...this.props} />
+        );
+      }
+    }, this);
+    return (
+      <div>
+        <span className="filter-heading">Teams</span>
+        <span className="filter-toggle" onClick={this.handleToggle}> {(this.state.expanded) ? '-':'+'} </span>
```

Comment:
The `filter-heading` and `filter-toggle` sections are duplicated 3x. Not sure if you want to convert them into a new component, and maybe something like this:
```html
<ToggleHeading label="Projects" handleToggle={this.handleToggle} expanded={this.state.expanded} />

<span className="filter-heading">{this.props.label}</span>
<span className="filter-toggle" onClick={this.props.handleToggle}>{this.state.expanded ? '-' : '+'}</span>
```

You could possibly also expand it to include the surrounding `<div>` and `<ul>{nodes}</ul></div>` since those seem boilerplate as well.